addons/stock/models/stock_quant.py

```diff
@@ -271,6 +271,12 @@ def create(self, vals_list):
                     # Merge quants later, to make sure one line = one record during batch import
                     quant = self._gather(product, location, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=True)
                 if lot_id:
+                    if self.env.context.get('import_file') and lot_id.product_id != product:
+                        lot_name = lot_id.name
+                        lot_id = self.env['stock.lot'].search([('product_id', '=', product.id), ('name', '=', lot_name)], limit=1)
+                        if not lot_id:
+                            lot_id = self.env['stock.lot'].create({'name': lot_name, 'product_id': product.id})
```

Comment:
set a company on the lot
(could use the one of the location or env's one as fallback)

Reply:
Done, thanks for the review !